pkg/webhook/resourcedeletionprotection/validating.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2023 The Karmada Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcedeletionprotection
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// ValidatingAdmission validates CronFederatedHPA object when creating/updating.
+type ValidatingAdmission struct {
+	Decoder *admission.Decoder
+}
+
+// Check if our ValidatingAdmission implements necessary interface
+var _ admission.Handler = &ValidatingAdmission{}
+
+// If a user assigns the ResourceDeletionProtectionLabelKey label to a specific resource,
+// and the value of this label is ResourceDeletionProtectionAlways, then deletion requests
+// for this resource will be denied.
+// In the current design, only the Value set to 'Always' will be protected,
+// while other values (including 'Never') will not receive any protection.
+// Additional options will be added here in the future.
+const (
+	ResourceDeletionProtectionLabelKey = "karmada.io/deletion-protected"
```

Comment:
I'd propose this label:
```
resourcetemplate.karmada.io/deletion-protected
```
Keep consistent with other labels/annotations that are designed for resource templates, like https://github.com/karmada-io/karmada/blob/f1f1a82dc73ae3828971fbdaf763c087f87f1291/pkg/apis/work/v1alpha2/well_known_constants.go#L73-L97

Reply:
It's a good idea, and I wasn't quite sure where to put it before.